### PR TITLE
Tune RBAC privileges to explicitly scope to only status permission

### PR DESCRIPTION
**Pull request by howardjohn**
This lows the privileges required to run Istiod.

Note the finalizer bit is removed since it is obsolete and part of a
removed feature.

### CODE REVIEW

File: manifests/charts/istio-control/istio-discovery/templates/clusterrole.yaml [hunk 1]
```diff
@@ -27,14 +27,22 @@ rules:
...
-    # TODO: should be on just */status but wildcard is not supported
-    resources: ["*"]
-
-  # Needed because status reporter sets the config map owner reference to the istiod pod
-  - apiGroups: [""]
-    verbs: ["update"]
-    resources: ["pods/finalizers"]
+    verbs: ["update", "patch"]
+    resources:
+    - authorizationpolicies/status
```
Comment (MorrisLaw) [inline manifests/charts/istio-control/istio-discovery/templates/clusterrole.yaml:32]: good change 👍

**Review by MorrisLaw — APPROVED** [review]

File: manifests/charts/istio-control/istio-discovery/templates/clusterrole.yaml [hunk 2]
```diff
@@ -27,14 +27,22 @@ rules:
...
   - apiGroups: ["config.istio.io", "security.istio.io", "networking.istio.io", "authentication.istio.io", "rbac.istio.io", "telemetry.istio.io", "extensions.istio.io"]
-    verbs: ["update"]
-    # TODO: should be on just */status but wildcard is not supported
-    resources: ["*"]
-
-  # Needed because status reporter sets the config map owner reference to the istiod pod
-  - apiGroups: [""]
-    verbs: ["update"]
-    resources: ["pods/finalizers"]
+    verbs: ["update", "patch"]
```
Comment (MorrisLaw) [inline manifests/charts/istio-control/istio-discovery/templates/clusterrole.yaml:30]: good change 👍

File: pilot/pkg/config/kube/gateway/deploymentcontroller.go [hunk 3]
```diff
@@ -547,7 +547,9 @@ func (d *DeploymentController) setGatewayControllerVersion(gws gateway.Gateway)
 		ControllerVersionAnnotation, ControllerVersion)
 
 	log.Debugf("applying %v", patch)
-	return d.patcher(gvr.KubernetesGateway, gws.GetName(), gws.GetNamespace(), []byte(patch))
+	// Use status RBAC so we do not require full Gateway write.
+	// `status` write can modify annotations, and we already need to write status anyway so we have the permission.
```
Comment (keithmattix) [inline pilot/pkg/config/kube/gateway/deploymentcontroller.go:551]: Looks like just a comment change; is that intended?

Comment (howardjohn) [reply pilot/pkg/config/kube/gateway/deploymentcontroller.go:551]: The last arg changed (`status`)

Comment (keithmattix) [reply pilot/pkg/config/kube/gateway/deploymentcontroller.go:551]: Ah, it indented to another line and I missed it; thanks

**Review by keithmattix — APPROVED** [review]
Integration tests pass so I'm confident we can still write/read whatever we need to with the lower permission

**Review by zirain — APPROVED** [review]

**Pull request merged**